<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Disruptive - DisruptiveError</title>
    <link rel="stylesheet" type="text/css" href="https://vegather.github.io/Disruptive/all.css" media="all" />
</head>
<body>
    <header>
        <a href="https://vegather.github.io/Disruptive/">
            <strong>
                Disruptive
            </strong>
            <span>Documentation</span>
        </a>
        <sup>Beta</sup>
    </header>

    <!--
    <form class="search">
        <input type="search" placeholder="Search" />
    </form>
    -->

    <nav>
        <div class="wrapper">
            <h2>On This Page</h2>
            <ol><li><a href="#relationships">Relationships</a><ul><li><a href="#relationships">Conforms To</a></li></ul></li><li><a href="#enumeration cases">Enumeration Cases</a><ul><li class="case"><a href="#disruptiveerror.serverunavailable">server​Unavailable</a></li><li class="case"><a href="#disruptiveerror.servererror">server​Error</a></li><li class="case"><a href="#disruptiveerror.badrequest">bad​Request</a></li><li class="case"><a href="#disruptiveerror.unauthorized">unauthorized</a></li><li class="case"><a href="#disruptiveerror.insufficientpermissions">insufficient​Permissions</a></li><li class="case"><a href="#disruptiveerror.notfound">not​Found</a></li><li class="case"><a href="#disruptiveerror.resourcealreadyexists">resource​Already​Exists</a></li><li class="case"><a href="#disruptiveerror.unknownerror">unknown​Error</a></li><li class="case"><a href="#disruptiveerror.loggedout">logged​Out</a></li></ul></li></ol>
        </div>
    </nav>

    <main>
        <article>
            <h1>
    <small>Enumeration</small>
    <code class="name">Disruptive​Error</code>
</h1>

<html><body><pre class="highlight"><code><span class="keyword">public</span> <span class="keyword">enum</span> <span class="type">DisruptiveError</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>An enumeration of all the possible errors that could occur by calling one of the endpoints.</p>

</div>
<section id="relationships">
    <h2 hidden>Relationships</h2>
        <figure>
    <?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.40.1 (20161225.0304)
 -->
<!-- Title: %3 Pages: 1 -->
<svg width="872pt" height="116pt"
 viewBox="0.00 0.00 872.00 116.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 112)">
<title>%3</title>
<polygon fill="#ffffff" stroke="transparent" points="-4,4 -4,-112 868,-112 868,4 -4,4"/>
<!-- DisruptiveError -->
<g id="node1" class="node enumeration current">
<title>DisruptiveError</title>
<g id="a_node1"><a xlink:href="https://vegather.github.io/Disruptive/DisruptiveError" xlink:title="DisruptiveError">
<path fill="none" stroke="#000000" stroke-width="3" d="M528,-108C528,-108 336,-108 336,-108 330,-108 324,-102 324,-96 324,-96 324,-84 324,-84 324,-78 330,-72 336,-72 336,-72 528,-72 528,-72 534,-72 540,-78 540,-84 540,-84 540,-96 540,-96 540,-102 534,-108 528,-108"/>
<text text-anchor="middle" x="432" y="-86.3" font-family="Menlo" font-size="14.00" fill="#000000">DisruptiveError</text>
</a>
</g>
</g>
<!-- Error -->
<g id="node2" class="node unknown">
<title>Error</title>
<path fill="none" stroke="#000000" d="M528,-36C528,-36 336,-36 336,-36 330,-36 324,-30 324,-24 324,-24 324,-12 324,-12 324,-6 330,0 336,0 336,0 528,0 528,0 534,0 540,-6 540,-12 540,-12 540,-24 540,-24 540,-30 534,-36 528,-36"/>
<text text-anchor="middle" x="432" y="-14.3" font-family="Menlo" font-size="14.00" fill="#000000">Error</text>
</g>
<!-- DisruptiveError&#45;&gt;Error -->
<g id="edge1" class="edge conformsTo">
<title>DisruptiveError&#45;&gt;Error</title>
<path fill="none" stroke="#000000" d="M432,-71.8314C432,-64.131 432,-54.9743 432,-46.4166"/>
<polygon fill="#000000" stroke="#000000" points="435.5001,-46.4132 432,-36.4133 428.5001,-46.4133 435.5001,-46.4132"/>
</g>
</g>
</svg>


    <figcaption hidden>Inheritance graph for DisruptiveError.</figcaption>
</figure>
        <h3>Conforms To</h3>
<dl>
    <dt class="unknown"><code>Error</code></dt>
</dl>
</section>
    <section id="enumeration cases">
        <h2>Enumeration Cases</h2>

        <div role="article" class="case" id="disruptiveerror.serverunavailable">
    <h3>
        <code>server​Unavailable</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">serverUnavailable</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Could not reach the server. Either the client is not connected
to the internet, or the server is currently offline.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.servererror">
    <h3>
        <code>server​Error</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">serverError</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Something went wrong on the server. This error comes from
a 500, 503, or 504 status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.badrequest">
    <h3>
        <code>bad​Request</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">badRequest</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Something went wrong with parsing the request on the
server. This error comes from a 400 status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.unauthorized">
    <h3>
        <code>unauthorized</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">unauthorized</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Either the client is not properly authorized, or the access
token has expired. This error comes from a 401 status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.insufficientpermissions">
    <h3>
        <code>insufficient​Permissions</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">insufficientPermissions</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>You attempted to access a resource that you don't
have access to. This error comes from either a 403
status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.notfound">
    <h3>
        <code>not​Found</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">notFound</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>You attempted to access a resource that you either don't
have access to, or doesn't exist. This error comes from
either a 403 or a 404 status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.resourcealreadyexists">
    <h3>
        <code>resource​Already​Exists</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">resourceAlreadyExists</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>You likely tried to create a resource that already exists.
This error comes from a 409 status code.</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.unknownerror">
    <h3>
        <code>unknown​Error</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">unknownError</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Something unexpected happened that could not be recovered from.
Check the logs for more information</p>

</div>
</div>
<div role="article" class="case" id="disruptiveerror.loggedout">
    <h3>
        <code>logged​Out</code>
    </h3>
    <html><body><pre class="highlight"><code><span class="keyword">case</span> <span class="variable">loggedOut</span></code></pre></body></html>
<div class="summary" role="doc-abstract">
    <p>Returned when the <code>authProvider</code> is currently logged out. Call
<code>login()</code> on the <code>authProvider</code> to log it back in.</p>

</div>
</div>
    </section>



        </article>
    </main>

    <footer>
        <p>
    Generated on <time datetime="2021-01-09T12:32:34+0000">January 9, 2021</time> using <a href="https://github.com/SwiftDocOrg/swift-doc">swift-doc</a> <span class="version">1.0.0-beta.5</span>.
</p>
    </footer>
</body>
</html>
